software/update.sh

```diff
@@ -25,62 +25,94 @@ then
   echo "ERROR: file $SECRETFILE missing: create this file with your PO.DAAC username and password, each in one single line."
   exit 3
 fi
-USERNAME=$(head -n1 $SECRETFILE)
-PASSWORD=$(tail -n1 $SECRETFILE)
+USERNAME=$(head -n1 "$SECRETFILE")
+PASSWORD=$(tail -n1 "$SECRETFILE")
 
 #download (if not already)
-[ ! -e $LOCAL_FILE ] && wget \
+[ ! -e "$LOCAL_FILE" ] && wget \
   --user=$USERNAME \
   --password=$PASSWORD \
   --no-directories \
   $REMOTE_FILE
 
-[[ ! "${@/--clean/}" == "$@" ]] && rm -fvr $SW_DIR
+[[ ! "${@/--clean/}" == "$@" ]] && rm -fvr "$SW_DIR"
 
 #decompress package (if not already)
-[ ! -d $SW_DIR ] && tar -x -C $LOCAL_DIR -f $LOCAL_FILE -v
+[ ! -d "$SW_DIR" ] && tar -x -C "$LOCAL_DIR" -f "$LOCAL_FILE" -v
 
+(find "$SW_DIR" -name Makefile) | cat  > temfile.txt
+filename=temfile.txt 
 #need to tweak the make files
-for i in $(find $SW_DIR -name Makefile)
```

Comment:
A better option to handle paths with blanks is:

`find $SW_DIR -name Makefile -print0 | while IFS= read -r -d '' i`

and making sure any reference to `$i` inside the loop is enclosed in quotes.

cf. https://askubuntu.com/questions/343727/filenames-with-spaces-breaking-for-loop-find-command

Reply:
Thanks.